### PR TITLE
docs: [IAC-2771]: Split provision and destroy workspace docs

### PR DESCRIPTION
**Pull request by richardblack-Harness**
## Description

* Separating Provision and Destroy workspace docs and adding interactive guides to help improve bounce rates.
* [JIRA](https://harness.atlassian.net/browse/IAC-2771)

PRs must meet these requirements to be merged:

- [x] Successful preview build.
- [x] Code owner review.
- [x] No merge conflicts.
- [x] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

File: docs/infra-as-code-management/pipelines/operations/destroy-workspaces.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+></iframe>
+</TabItem>
+<TabItem value="Step-by-Step" label="Step-by-Step">
+Follow these instructions to provision a workspace within the Harness Visual Editor.
+
+1. In the Harness project pane, select **Pipelines**.
+2. Click **+ Create a Pipeline** to set up a new pipeline.
+3. Click **Add Stage** and select **Infrastructure** from the **Select Stage Type** pane.
+4. Enter a stage name, optional description, and tag, then select **Set Up Stage**.
+5. On the **Workspace** tab, select an existing workspace or click **+ Create New Workspace** to create one. Note: If set to `runtime input`, the workspace can be specified during execution.
```
Comment (scottyw-harness) [inline docs/infra-as-code-management/pipelines/operations/destroy-workspaces.md:34]: If they're running a destroy they must already have a workspace so I think we're saying "select your workspace" only

File: docs/infra-as-code-management/pipelines/operations/destroy-workspaces.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+6. Go to the **Execution** tab. Under **Common Operations**, select **Destroy** to configure the workspace destruction process.
+7. Select **Use Strategy** to automatically add `init`, `plan`, and `destroy` steps, or customize the pipeline by adding steps manually.
+8. Click **Save** and then **Run Pipeline** to execute.
+
+:::note
+The plan-destroy step generates a Terraform plan. This Terraform plan is accessible to all the steps after the IaCM Terraform plan, and can be referenced using the expression `<+pipeline.stages.STAGE_ID.spec.execution.steps.STEP_ID.output.outputVariables.parsedPlan>`.
+:::
+</TabItem>
+<TabItem value="YAML" label="YAML">
+To provision a workspace via YAML, use the template below. Replace bracketed placeholders as needed.
```
Comment (scottyw-harness) [inline docs/infra-as-code-management/pipelines/operations/destroy-workspaces.md:44]: Should maybe be "To destroy a workspace"

File: docs/infra-as-code-management/pipelines/operations/destroy-workspaces.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                  name: destroy
+                  identifier: destroy
+                  spec:
+                    command: destroy
+                  description: destroy
+                  timeout: 5m
+        tags: {}
+```
+:::note
+Notice the three steps to apply your `init`, `plan` and `destroy` commands.
```
Comment (scottyw-harness) [inline docs/infra-as-code-management/pipelines/operations/destroy-workspaces.md:94]: Maybe don't use the verb "apply" which means something special in TF. Even like "run" or "execute".

File: docs/infra-as-code-management/pipelines/operations/destroy-workspaces.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+:::note
+Notice the three steps to apply your `init`, `plan` and `destroy` commands. 
+:::
+</TabItem>
+</Tabs>
+---
+After the plan is created, view resources and Terraform outputs on the **Resources** tab. Check which resources will be added, modified, or removed.
+
+:::tip pipeline chaining
+Currently, IaCM doesn't support multiple Workspaces in the same pipeline. If you'd like to provision several Workspaces altogether, consider using [Pipeline Chaining](https://developer.harness.io/docs/platform/pipelines/pipeline-chaining/).
```
Comment (scottyw-harness) [inline docs/infra-as-code-management/pipelines/operations/destroy-workspaces.md:102]: I think we fixed this and now support multiple. @allofthesepeople would know for sure.

File: docs/infra-as-code-management/pipelines/operations/destroy-workspaces.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+```yaml
+pipeline:
+  name: <<PIPELINE NAME>>
+  identifier: <<PIPELINE IDENTIFIER>>
+  projectIdentifier: <<PROJECT IDENTIFIER>>
+  orgIdentifier: <<ORG IDENTIFER>>
+  tags: {}
+  stages:
+    - stage:
+        name: iacstage
```
Comment (scottyw-harness) [inline docs/infra-as-code-management/pipelines/operations/destroy-workspaces.md:55]: The "apply" example names these nicely:

```
        name: Provision Stage
        identifier: provision_stage
```

File: docs/infra-as-code-management/pipelines/operations/provision-workspace.md [hunk 6]
```diff
@@ -1,197 +1,100 @@
...
-                  spec:
-                    command: destroy
-                  description: destroy
-                  timeout: 5m
-        tags: {}
-```
-
-
+:::note
+Notice the three steps to apply your `init`, `plan` and `apply` commands.
```
Comment (scottyw-harness) [inline docs/infra-as-code-management/pipelines/operations/provision-workspace.md:91]: Ditto "apply", same as above

File: docs/infra-as-code-management/pipelines/operations/provision-workspace.md [hunk 7]
```diff
@@ -1,197 +1,100 @@
...
 </Tabs>
+---
+After the plan is created, view resources and Terraform outputs on the **Resources** tab. Check which resources will be added, modified, or removed.
 
-
-:::tip
-Currently, IaCM doesn't support multiple Workspaces in the same pipeline. If you'd like to provision several Workspaces altogether, consider using [Pipeline Chaining](https://developer.harness.io/docs/platform/pipelines/pipeline-chaining/)
-:::
+:::tip pipeline chaining
+Currently, IaCM doesn't support multiple Workspaces in the same pipeline. If you'd like to provision several Workspaces altogether, consider using [Pipeline Chaining](https://developer.harness.io/docs/platform/pipelines/pipeline-chaining/).
```
Comment (scottyw-harness) [inline docs/infra-as-code-management/pipelines/operations/provision-workspace.md:99]: Same

**Review by scottyw-harness — APPROVED** [review]

File: docs/infra-as-code-management/pipelines/operations/destroy-workspaces.md (hunk 1 above)
Comment (richardblack-Harness) [reply docs/infra-as-code-management/pipelines/operations/destroy-workspaces.md:34]: done

File: docs/infra-as-code-management/pipelines/operations/destroy-workspaces.md (hunk 2 above)
Comment (richardblack-Harness) [reply docs/infra-as-code-management/pipelines/operations/destroy-workspaces.md:44]: done

File: docs/infra-as-code-management/pipelines/operations/destroy-workspaces.md (hunk 5 above)
Comment (richardblack-Harness) [reply docs/infra-as-code-management/pipelines/operations/destroy-workspaces.md:55]: done

File: docs/infra-as-code-management/pipelines/operations/destroy-workspaces.md (hunk 3 above)
Comment (richardblack-Harness) [reply docs/infra-as-code-management/pipelines/operations/destroy-workspaces.md:94]: Yes, Execute I think works best.

File: docs/infra-as-code-management/pipelines/operations/provision-workspace.md (hunk 6 above)
Comment (richardblack-Harness) [reply docs/infra-as-code-management/pipelines/operations/provision-workspace.md:91]: done

**Review by scottyw-harness — APPROVED** [review]

**Pull request merged**